### PR TITLE
Add QP complementarity test

### PR DESCRIPTION
**Pull request by frapac**
I stay at your disposal if you want to add other MPEC tests inside MOI

### CODE REVIEW

**Comment by frapac** [comment]
I understand. I still have to bridge Knitro then.

**Comment by odow** [comment]
You should test the test with a `mock` optimizer. See my PR for an example.

**Comment by frapac** [comment]
Just a question, however. You mentioned earlier:
> KNITRO.jl, which will implement VectorOfVariables-in-Complements

Shouldn't we test this feature as well?

File: test/Test/nlp.jl [hunk 1]
```diff
@@ -16,4 +16,5 @@ const MOI = MathOptInterface
         )
     )
     MOI.Test.test_linear_mixed_complementarity(mock, config)
+    MOI.Test.test_qp_mixed_complementarity(mock, config)
```
Comment (odow) [inline test/Test/nlp.jl]: You need to set the solution in the mock optimizer like I do here:
https://github.com/JuliaOpt/MathOptInterface.jl/pull/913/files#diff-00b88401713e1c20041db74f14c5ccd1R12-R17

Comment (blegat) [inline test/Test/nlp.jl]: You probably need to add a call to `MOI.Utilities.set_mock_optimize!` as above to set the optimal solution.

File: src/Test/nlp.jl [hunk 2]
```diff
@@ -371,16 +371,108 @@ function test_linear_mixed_complementarity(model::MOI.ModelLike, config::TestCon
...
+        @test isapprox(
+            x_val, [1., 0., 3.5, 0., 0., 0., 3., 6], atol = config.atol, rtol = config.rtol
+        )
+        @test MOI.get(model, MOI.ObjectiveValue()) ≈ 17.0 atol=config.atol rtol=config.rtol
+    end
 end
 
 const complementaritytests = Dict(
     "linear_mixed_complementarity" => test_linear_mixed_complementarity,
+    "qp_mixed_complementarity" => test_linear_mixed_complementarity,
```
Comment (odow) [inline src/Test/nlp.jl]: Should be `=> test_qp_mixed_complementarity`

File: src/Test/nlp.jl [hunk 3]
```diff
@@ -371,16 +371,108 @@ function test_linear_mixed_complementarity(model::MOI.ModelLike, config::TestCon
...
+        x0, x1, x2, x3, x4, x5, x6, x7 >= 0
+
+```
+"""
+function test_qp_mixed_complementarity(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    x = MOI.add_variables(model, 8)
+    MOI.set.(model, MOI.VariablePrimalStart(), x, 0.0)
+    for i in 1:8
+        MOI.add_constraint(model, MOI.SingleVariable(x[i]), MOI.GreaterThan(0.))
```
Comment (odow) [inline src/Test/nlp.jl]: Please add a trailing `0` to `0.`. There are also multiple cases of this further down.

File: src/Test/nlp.jl (hunk 2 above)
Comment (blegat) [reply src/Test/nlp.jl]: That's the reason the tests are passing without `MOIU.set_mock_optimize!`

**Pull request merged**